src/engine/bufferscalers/enginebufferscalelinear.cpp

```diff
@@ -289,12 +288,11 @@ SINT EngineBufferScaleLinear::do_scale(CSAMPLE* buf, SINT buf_size) {
                         rate_new == 0 ? rate_old : rate_new,
                         m_bufferInt, samples_to_read);
 
-                if (m_bufferIntSize == 0) {
-                    if (++read_failed_count > 1) {
-                        break;
-                    } else {
-                        continue;
-                    }
+                VERIFY_OR_DEBUG_ASSERT(m_bufferIntSize != 0) {
+                    // getNextSamples() must never fail,
+                    // using silence as a last resort
+                    SampleUtil::clear(m_bufferInt, samples_to_read);
+                    m_bufferIntSize = getOutputSignal().samples2frames(samples_to_read);
```

Comment:
for example 
> when moving a loop causes the play position to be moved along

samples_to_read may be odd and thus hit
https://github.com/mixxxdj/mixxx/blob/5f5cd28b3f8b471f5f2ca271e1ecacf8f2b1a659/src/audio/signalinfo.h#L40-L44

SampleUtil::roundPlayPosToFrameStart ?

Reply:
Tbh I don't understand why you used `samples2frames` here.
FWIW if I just do
```
if (m_bufferIntSize == 0) {
    // getNextSamples() must never fail,
    // using silence as a last resort
    SampleUtil::clear(m_bufferInt, samples_to_read);
    m_bufferIntSize = samples_to_read;
}
```
that's probably what you had in mind, but it causes #11381.

Though, if I remove this debug assert altogether, i.e. just continue, the loop move use case is covered and #11381 is fixed!